sycl/source/detail/pi_opencl.cpp

```diff
@@ -78,6 +78,43 @@ pi_result OCL(piextDeviceSelectBinary)(
   return PI_SUCCESS;
 }
 
+pi_result OCL(piQueueCreate)(pi_context context, pi_device device,
+                             cl_command_queue_properties properties,
+                             pi_queue *queue) {
+  size_t devVerSize;
+  cl_int ret_err;
+  ret_err = clGetDeviceInfo(pi_cast<cl_device_id>(device), CL_DEVICE_VERSION, 0,
+                            NULL, &devVerSize);
+
+  if (ret_err != CL_SUCCESS)
+    return pi_cast<pi_result>(ret_err);
+
+  std::string devVer(devVerSize, '\0');
+  clGetDeviceInfo(pi_cast<cl_device_id>(device), CL_DEVICE_VERSION, devVerSize,
+                  &devVer.front(), NULL);
```

Comment:
`devVer.data()` instead

Reply:
going to leave it as .front() for now as the Upstream implementation will probably need to remain .front() and changing this will just cause some pull request conflicts.

Reply:
Actually you are right, it does not work before C++17. Since `devVer.front()` is a reference according to the C++ specification it works and is even shorten than `&*devVer.begin()`